src/App.tsx

```diff
@@ -1,26 +1,411 @@
+                />
+              </label>
+
+              {ActiveChangeId !== todo.id ? (
+                <span
+                  data-cy="TodoTitle"
+                  className="todo__title"
+                  onDoubleClick={() => {
+                    setActiveChangeId(todo.id);
+                    setEditedTodo(todo.title);
+                  }}
+                >
+                  {todo.title}
+                </span>
+              ) : (
+                <form onSubmit={event => event.preventDefault()}>
+                  <input
+                    data-cy="TodoTitleField"
+                    type="text"
+                    className="todo__title-field"
+                    placeholder={todo.title}
+                    value={editedTodo}
+                    onChange={event => setEditedTodo(event.target.value)}
+                    onBlur={() => loseFocus(0)}
+                    onKeyDown={hendleEdit}
+                    autoFocus
+                  />
+                </form>
+              )}
+
+              {/* Remove button appears only on hover */}
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+                onClick={() => handleDelete(todo.id)}
+              >
+                ×
+              </button>
 
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+              {/* overlay will cover the todo while it is being deleted or updated */}
+              <div
+                data-cy="TodoLoader"
+                className={`modal overlay ${cn({ 'is-active': deletingId?.includes(todo.id) })}`}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          ))}
+          {/* This todo is an active todo */}
+
+          {/* This todo is in loadind state */}
+          {Boolean(tempTodo) && (
+            <div data-cy="Todo" className="todo">
+              <label className="todo__status-label">
+                <input
+                  data-cy="TodoStatus"
+                  type="checkbox"
+                  className="todo__status"
+                />
+              </label>
+
+              <span data-cy="TodoTitle" className="todo__title">
+                {zeroIdElement.title}
+              </span>
+
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+              >
+                ×
+              </button>
+              <div data-cy="TodoLoader" className="modal overlay is-active">
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          )}
+        </section>
+
+        {/* Hide the footer if there are no todos */}
+        {!!todos.length && (
```

Comment:
```suggestion
        {Boolen(todos.length) && (
```
write code in the same style